src/evaluation/evaluator.h

```diff
@@ -1,312 +1,271 @@
+    uint8_t searcherId;
+    movegen::Move pvMove;
+    uint8_t searchedDepth;
+};
 
-        /* we're dependent on the hash table - ensure it's initialized! */
-        if (engine::TtHashTable::getSizeMb() == 0) {
-            engine::TtHashTable::setSizeMb(s_defaultTtHashTableSizeMb);
-        }
+template<uint8_t maxSize>
+class MoveVoteMap {
 
-        return scanForBestMove(depthInput.value_or(s_maxSearchDepth), board);
+    using MoveVotePair = std::pair<movegen::Move, uint8_t>;
+
+public:
+    MoveVoteMap() = default;
+
+    MoveVoteMap(std::initializer_list<MoveVotePair> init)
+    {
+        assert(init.size() < maxSize);
+
+        for (auto& entry : init) {
+            m_entries.at(m_size++) = std::move(entry);
+        }
     }
 
-    constexpr std::optional<movegen::Move> getPonderMove() const
+    constexpr uint8_t& at(const movegen::Move& ext_move)
     {
-        if (m_moveOrdering.pvTable().size() > 1) {
-            return m_moveOrdering.pvTable()[1];
+        for (auto& [move, vote] : m_entries) {
+            if (move == ext_move) {
+                return vote;
+            }
         }
+        assert(false);
 
-        return std::nullopt;
+        return m_entries.at(0).second;
     }
 
-    /* TODO: ensure time handler is stopped as well - not just assumed stopped. This is racy */
-    constexpr void stop()
+    constexpr bool count(const movegen::Move& ext_move)
     {
-        m_isStopped = true;
+        for (auto& [move, vote] : m_entries) {
+            if (move == ext_move) {
+                return true;
+            }
+        }
+        return false;
     }
 
-    constexpr uint64_t getNodes() const
+    constexpr void insert(movegen::Move move, uint8_t vote)
     {
-        return m_nodes;
+        assert(m_size < maxSize);
+        m_entries.at(m_size++) = MoveVotePair(move, vote);
     }
 
-    constexpr void printEvaluation(const BitBoard& board, std::optional<uint8_t> depthInput = std::nullopt)
+    constexpr void clear()
     {
-        resetTiming(); /* to reset nodes etc but not tables */
+        for (uint8_t i = 0; i < m_size; i++) {
+            m_entries.at(i) = MoveVotePair {};
+        }
+        m_size = 0;
+    }
 
-        uint8_t depth = depthInput.value_or(5);
+    auto* begin() { return m_entries.begin(); }
 
-        fmt::println("");
+    auto* end() { return m_entries.begin() + m_size; }
 
-        movegen::ValidMoves captures;
-        engine::getAllMoves<movegen::MoveCapture>(board, captures);
-        if (captures.count()) {
-            m_moveOrdering.sortMoves(board, captures, m_ply);
+private:
+    std::array<MoveVotePair, maxSize> m_entries {};
+    uint8_t m_size {};
+};
 
-            fmt::print("Captures[{}]: ", captures.count());
-            for (const auto& move : captures) {
-                fmt::print("{} [{}]  ", move, m_moveOrdering.moveScore(board, move, 0));
-            }
-            fmt::print("\n\n");
-        }
+/* Base class for searcher */
+class Searcher {
```

Comment:
I think this belongs in its own file :)
(would also have made rebasing easier)

Reply:
Done. It's handy to know for the future that a tricky rebase can be a hint towards file seperation :)